dmocks/dynamic.d

```diff
@@ -163,4 +171,10 @@ version (DMocksTest) {
         d.get!D;
     }
     +/
+
+    // Compiles with a const object with the mutable toString.
+    unittest
+    {
+        auto d = dynamic(new const Object);
```

Comment:
"immutable" will be casted away. Although I'm not sure if casting away the immutability is a good idea. Casting away the costness can be ok if the object wasn't originally const (but was passed to a function that accepts a const reference for example). Casting away immutability can cause undefined behavior if this is not the only reference.
But since we're assuming toString() doesn't modify the object, it could work.